src/posit/connect/urls.py

```diff
@@ -5,23 +5,18 @@
 from urllib.parse import urlsplit, urlunsplit
 
 
-def append_path(url: str, path: str) -> str:
+def validate(url: str) -> None:
     """
-    Appends a path to the end of a URL.
+    Check if the given URL is valid.
 
     Args:
-        url (str): The URL to append the path to.
-        path (str): The path to append to the URL.
+        url (str): The URL to be validated.
 
     Returns:
-        str: The modified URL with the appended path.
+        bool: True if the URL is valid, False otherwise.
 
     Raises:
-        ValueError: If the URL does not specify a scheme, is not absolute, or does not end with "/__api__".
-
-    Example:
-        >>> append_path("http://example.com", "api")
-        'http://example.com/__api__/api'
+        ValueError: If the URL is missing a scheme, is not absolute, or does not end with '/__api__'.
```

Comment:
I don't think we should require users to add `__api__` to their server URLs. Convention everywhere in our [docs](https://docs.posit.co/connect/cookbook/), and presumably in the env vars we inject into content, is that `__api__` is not included in the URL you provide:

> Your Connect server URL is the same URL you use to access the Connect dashboard, minus the connect path. If you access the dashboard at https://rsc.company.com/connect/, the server URL is https://rsc.company.com/.

We should add it if it is not present, IMO.

Reply:
Good call. I'll add that back in.